frontend/include/chpl/framework/query-impl.h

```diff
@@ -395,10 +406,17 @@ Context::updateResultForQueryMapR(QueryMap<ResultType, ArgTs...>* queryMap,
   bool initialResult = (r->lastChanged == -1);
   auto currentRevision = this->currentRevisionNumber;
 
+  // If recursion errors happened at the time the result is being saved,
+  // the query is 'poisoned' by recursion and we should store the default result.
+  if (!r->recursionErrors.empty()) {
+    ResultType dummyValue {};
```

Comment:
Isn't

```
    ResultType dummyValue;
```

simpler and sufficient here?

Reply:
that's what I wrote, but it's not equivalent! 

```
int myInt; // uninitialized
int myInt {}; // initialized to zero.
```

As you can imagine, we need the latter :)

Reply:
Very good, might be worth a comment for anybody maintaining it who isn't thinking of the `int` case.

Reply:
I've added a comment 👍